# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.h
# Commit: 5a94b945a3ad
# Full Hash: 5a94b945a3ad6bf603d47c07788b7a70ba651320
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 4: Implement ActorDestroy r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227689
# ==============================================================================

diff -r 984f6e687ff5 -r 5a94b945a3ad dom/notification/NotificationParent.h
--- a/dom/notification/NotificationParent.h	Thu Nov 07 17:32:04 2024 +0000
+++ b/dom/notification/NotificationParent.h	Thu Nov 07 17:32:05 2024 +0000
@@ -35,21 +35,32 @@
         mOptions(aOptions) {};
 
   IPCResult RecvShow(ShowResolver&& aResolver);
-  IPCResult RecvClose();
 
   nsresult BindToMainThread(
       Endpoint<PNotificationParent>&& aParentEndpoint,
       PBackgroundParent::CreateNotificationParentResolver&& aResolver);
 
+  void ActorDestroy(ActorDestroyReason aWhy) override;
+
  private:
   ~NotificationParent() = default;
 
+  void GetAlertName(nsAString& aRetval) {
+    if (mAlertName.IsEmpty()) {
+      MaybeInitAlertName();
+    }
+    aRetval = mAlertName;
+  }
+  void MaybeInitAlertName();
+
   NotNull<nsCOMPtr<nsIPrincipal>> mPrincipal;
   NotNull<nsCOMPtr<nsIPrincipal>> mEffectiveStoragePrincipal;
   bool mIsSecureContext;
   nsString mId;
   nsString mScope;
   IPCNotificationOptions mOptions;
+
+  nsString mAlertName;
 };
 
 }  // namespace mozilla::dom::notification
